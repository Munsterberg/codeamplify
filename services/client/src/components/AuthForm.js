import React from 'react';
import { string, object } from 'prop-types';

const propTypes = {
  formType: string,
  formData: object
};

const AuthForm = ({ formType, formData }) => {
  return (
    <div>
      <h1>{ formType }</h1>
      <hr/>
      <form>
        {
          formType === 'Register' &&
            <div className="form-group">
              <input
                type="email"
                name="email"
                className="form-control input-lg"
                placeholder="Email Address"
                required
                value={formData.email}
                onChange={() => {}}
              />
            </div>
        }
        <div className="form-group">
          <input
            type="text"
            name="username"
            className="form-control input-lg"
            placeholder="Username"
            required
            value={formData.username}
            onChange={() => {}}
          />
        </div>
        <div className="form-group">
          <input
            type="password"
            name="password"
            className="form-control input-lg"
            placeholder="Password"
            required
            value={formData.password}
            onChange={() => {}}
          />
        </div>
        <input
          type="submit"
          className="btn btn-primary btn-lg btn-block"
          value="Submit"
        />
      </form>
    </div>
  );
};

AuthForm.propTypes = propTypes;


export default AuthForm;