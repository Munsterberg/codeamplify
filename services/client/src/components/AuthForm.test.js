import React from 'react';
import { shallow } from 'enzyme';

import AuthForm from './AuthForm';

describe('<AuthForm />', () => {
  const formData = {
    username: '',
    email: '',
    password: ''
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AuthForm formData={formData} formType={'Register'} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render h1 for register form', () => {
    const h1Tag = wrapper.find('h1');
    expect(h1Tag.length).toBe(1);
    expect(h1Tag.get(0).props.children).toBe('Register');
  });

  it('should render register form components', () => {
    const formGroup = wrapper.find('.form-group');
    expect(formGroup.length).toBe(3);
  });

  it('should render h1 for login form', () => {
    wrapper = shallow(<AuthForm formData={formData} formType={'Login'} />);
    const h1Tag = wrapper.find('h1');
    expect(h1Tag.length).toBe(1);
    expect(h1Tag.get(0).props.children).toBe('Login');
  });

  it('should render login form components', () => {
    wrapper = shallow(<AuthForm formData={formData} formType={'Login'} />);
    const formGroup = wrapper.find('.form-group');
    expect(formGroup.length).toBe(2);
  });
});