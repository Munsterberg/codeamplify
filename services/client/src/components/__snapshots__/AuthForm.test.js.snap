// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> should render correctly 1`] = `
<div>
  <h1>
    Register
  </h1>
  <hr />
  <form>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-primary btn-lg btn-block"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
